### DISCUSSION TITLE
如何从预合成的compositionli获取ImageLayer图层?

### DISCUSSION THREAD

**Discussion by cgeffect** [Tencent/libpag#2011] [Q&A] [answered]
有三个图层, 时间线分别是0-1, 1-2, 2-3(秒), 每一个图层使用了预合成组合成, 通过PAG接口解析出来的图层的开始时间都是0, 但是预合成的时间是正确的0-1, 1-2, 2-3. 通过PAGViewer工具的文件结构可以看到结构是

```
compositions:[
   	VectorComposition:{
    	 layers:[
         	PreComposeLayer:{
            	VectorComposition: {
            		layers:[
            			ImageLayer: {}
            		]
            	}
         	}
      	]
   }
]
```
可以看到最下面的ImageLayer, 但是通过Api接口只能拿到ImageLayer图层, 但是因为这个图层的父图层是PreComposeLayer, 所以ImageLayer图层的开始时间变成了0. 因为现在ImageLayer的图层开始时间都是0, 所所以要通过预合成图层去获取在整个时间线里的时间, 该怎么获取? 我自己测试, 拿到预合成图层之后, 预合成下面的composition属性并不是VectorComposition类型. 无法获取预合成下面的自图层.

是否是PAGViewer显示的数据结构和libpag里直接的接口不同, 通过libpag是无法获取到PAGViewer显示的数据结构?

**Comment by kevingpqi123** [comment, marked as answer]
此类问题建议系统看下 API 文档：
Converts the time from the PAGLayer’s (local) timeline to the [PAGSurface](https://pag.art/apis/ios/Classes/PAGSurface.html) (global) timeline. The time is in microseconds.

- (int64_t)localTimeToGlobal:(int64_t)localTime
Declared In
PAGLayer.h
